src/App.js

```diff
@@ -2,62 +2,102 @@ import './App.css';
 import VehicleList from './VehicleList';
 import TrafficLight from './TrafficLight';
 
+import React, { useState } from 'react';
+
+
 function App() {
   // track the following state with a few useState hooks:
   // lightColor should be a string that starts out as 'red'
+  const [lightColor, setLightColor] = useState('red');
   // lizardSize should be a number that starts out as 10
+  const [lizardSize, setLizardSize] = useState(10);
   // alienSize should be a number that starts out as 10
+  const [alienSize, setAlienSize] = useState(10);
   // traffic is complicated. It should be an array of strings that starts out as ['car', 'truck']
+  const [vehicleParade, setVehicleParade] = useState(['car', 'truck']);
+
+  function handleAlienEat() {
+    setAlienSize(alienSize + 1);
+  }
+  function handleAlienAttack() {
+    setLizardSize(lizardSize - 1);
+  }
+
+  function handleLizardEat() {
+    setLizardSize(lizardSize + 1);
+  }
+  function handleLizardAttack() {
+    setAlienSize(alienSize - 1);
+  }
+
+
+
+  function handleLightRed() {
+    setLightColor('red');
+  }
+  function handleLightYellow() {
+    setLightColor('yellow');
+  }
+  function handleLightGreen() {
+    setLightColor('green');
+  }
+
+  function handleAddVehicle(vehicle) {
+    vehicleParade.push(vehicle);
+
+    setVehicleParade([...vehicleParade]);
```

Comment:
this works, but you could also avoid doing the push on line 46 and just have line 48 and do this instead: `setVehicleParade([...vehicleParade, vehicle];`